torch/_functorch/_aot_autograd/jit_compile_runtime_wrappers.py

```diff
@@ -446,6 +801,8 @@ def aot_dispatch_autograd(
                 + num_tokens  # See Note [Side-Effectful Tokens in AOTAutograd]
             )
             fake_mode = detect_fake_mode()
+            fx_g = run_joint_graph_passes_on_hops(fx_g, joint_inputs, aot_config)
```

Comment:
fwiw, I don't think the "delayed generation of joint graph" needs to happen in this PR, but it's technically silent correctness so we probably don't want to leave it out for too long

Reply:
Collecting gaps here - https://docs.google.com/document/d/1B2OT867pmgKphChFw9Yus6tUNndCFTlGthT-AF7ScV0/edit?tab=t.0